src/metrics/metrics.ts

```diff
@@ -0,0 +1,214 @@
+import type {LogSink} from '@rocicorp/logger';
+
+export interface ReporterOptions {
+  datadogApiKey: string;
+  metrics: Metrics;
+  // intervalSecs defaults to 2 minutes.
+  intervalSecs?: number;
```

Comment:
Optional props should generally also allow undefined
```suggestion
  intervalSecs?: number | undefined;
```

Reply:
Hmm. How could it *not* allow undefined? Like why isn't the | undefined implicit? Or said another way, IF the property is optional, why is it not an error for the type to exclude undefined?

(Fixed, but still asking)

Reply:
https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes explains this in more details

Reply:
"colorThemeOverride: undefined is not the same as colorThemeOverride not being defined" is the only sentence you need in order to know how crazy javascript drives me sometimes